mlflow/pyfunc/__init__.py

```diff
@@ -1999,6 +2009,25 @@ def predict(model_input: List[str]) -> List[str]:
                 python_model, input_arg_index, input_example=input_example
             ):
                 mlflow_model.signature = signature
+        elif isinstance(python_model, ChatModel):
+            mlflow_model.signature = ModelSignature(
+                CHAT_MODEL_INPUT_SCHEMA,
+                CHAT_MODEL_OUTPUT_SCHEMA,
+            )
+            input_example = CHAT_MODEL_INPUT_EXAMPLE
+
+            # perform output validation and throw if
+            # output is not coercable to ChatResponse
+            messages = [ChatMessage(**m) for m in input_example["messages"]]
+            params = ChatParams(**{k: v for k, v in input_example.items() if k != "messages"})
+            output = python_model.predict(None, messages, params)
```

Comment:
is it a problem to perform inference during saving? i saw we do it when trying to infer output signature, but since this is kind of an LLM-specific API, inference can be kind of expensive. the input example specified `max_tokens=10`, so hopefully it isn't too bad.

if it is a concern, maybe we can just skip output validation entirely.

Reply:
I think there are some risks:

1. It may take a while (e.g. a few seconds) for the API request to finish.
2. No guarantee that the LLM service is healthy. If OpenAI is down, this line would throw.

Reply:
+1 We shouldn't predict while saving the model, the error message would be confusing.

Reply:
from discussion offline, we'll keep the predict since we do it in transformers/other places already for output signature inference. i'll do some more testing here to make sure it's not a confusing experience